File: Ex05/app/src/main/java/com/example/ex05/MainActivity.java
package com.example.ex05;

import android.graphics.Color;
import android.os.Bundle;
import android.view.View;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });


        View _colorLayout = findViewById(R.id.colorView);


        findViewById(R.id.btnRed).setOnClickListener(v -> {

            //_colorLayout.setBackgroundColor(0xffff0000);
            _colorLayout.setBackgroundColor(Color.rgb(255,0,0));

        });

        findViewById(R.id.btnGreen).setOnClickListener(v -> {

            //_colorLayout.setBackgroundColor(0xff00ff00);
            _colorLayout.setBackgroundColor(Color.rgb(0,255,0));

        });

    }
}

File: BleunoTest/app/src/main/java/com/example/bleunotest/MainActivity.kt
package com.example.bleunotest

import android.bluetooth.BluetoothAdapter
import android.bluetooth.BluetoothManager
import android.bluetooth.le.BluetoothLeScanner
import android.content.Context
import android.os.Bundle
import android.util.Log
import android.widget.Button
import androidx.activity.enableEdgeToEdge
import androidx.activity.result.ActivityResultLauncher
import androidx.activity.result.contract.ActivityResultContracts
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {

    private lateinit var btnScan: Button

    private lateinit var bluetoothAdapter: BluetoothAdapter
    private lateinit var bluetoothManager: BluetoothManager
    private lateinit var bluetoothLeScanner : BluetoothLeScanner

    private lateinit var permissionLauncher : ActivityResultLauncher<Array<String>>

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        bluetoothManager = getSystemService(Context.BLUETOOTH_SERVICE) as BluetoothManager
        bluetoothAdapter = bluetoothManager.adapter
        bluetoothLeScanner = bluetoothAdapter.bluetoothLeScanner

        permissionLauncher = registerForActivityResult(
            ActivityResultContracts.RequestMultiplePermissions()
        ) { permissions ->

            var allPermissionGranted = true

            permissions.entries.forEach { entry ->
                val permissionName = entry.key
                val isGranted = entry.value
                if (!isGranted) {
                    allPermissionGranted = false
                    Log.d("MainActivity", "permission denied: $permissionName")
                }
                else {
                    Log.d("MainActivity", "permission granted: $permissionName")
                }
            }

            if (allPermissionGranted) {
                Log.d("MainActivity", "all permission granted")
            }
            else {
                Log.d("MainActivity", "some permission denied")
            }
        }



        btnScan = findViewById(R.id.btnScan)

        btnScan.setOnClickListener { v->
            //TODO

            Log.d("MainActivity", "test")
        }

    }
}